ui/wobbly.js

```diff
@@ -307,18 +307,34 @@ function enableWobblyFx(wm) {
         SETTINGS_HANDLER = Settings.connect('changed::wobbly-effect', actionWobblyEffectSetting);
         actionWobblyEffectSetting(Settings, 'wobbly-effect');
     });
+
+    return GLib.SOURCE_REMOVE;
+}
+
+function enableWobblyFx(wm) {
+    initAnimationsDbus(wm);
 }
 
 function disableWobblyFx(wm) {
     Settings.disconnect(SETTINGS_HANDLER);
 
     getAnimatableWindowActors().forEach(actor => {
-        actor._animatableSurface = null;
+        if (actor._animatableSurface) {
+            wm._animationsServer.unregister_surface(actor._animatableSurface);
+            actor._animatableSurface = null;
+        }
     });
 
     if (wm._wobblyEffect) {
         wm._wobblyEffect.destroy();
         wm._wobblyEffect = null;
     }
-    wm._animationsServer = null;
+    if (wm._animationsManager) {
+        wm._animationsManager.unexport();
```

Comment:
Same as above, is this still needed?

Reply:
Same as above, I think it can’t hurt. (Demonstrably, it doesn’t hurt, because the code works.)